### DISCUSSION TITLE
Next-Auth + next-intl breaks fetch('/api')

### DISCUSSION THREAD

**Discussion by Arce0406** [amannn/next-intl#419] [Q&A]
I am using **Next.js 13**.

When using **Next-Auth** with **Next.js 13**, we will need a Route Handlers at the following paths
`/app/api/auth/[.... .nextauth]/route.ts`

At the same time, we also need to set the middleware.ts

When we use both **Next-Auth** and **next-intl**, the setup becomes more complicated, but there is an official tutorial and example that shows us how to do it.
[Next.js 13 middleware for i18n routing](https://next-intl-docs.vercel.app/docs/routing/middleware)
[Sample from amannn](https://github.com/amannn/next-intl)

But I've seen very few explanations of this problem below, and I'm not sure if it's just my way of setting things up that's wrong, or if everyone else has figured out a way to solve this problem.

Here is my problem：
I am using **Next-Auth** with **next-intl (App Router (Server Components))**
The folder structure is
`src/app/[locale]`

1. `src/app/[locale]/api`
If we put `app/api` into `app/[locale]/api`, then the general API will work e.g. `fetch('/api/test')` .
But it will cause next-auth's authentication api `/app/[locale]/api/auth/[...nextauth].ts` failed.
You will not be able to use the signin or signout function.

2. `src/app/api`
If we use `app/api`, then next-auth's authentication api `/app/api/auth/[...nextauth].ts` works fine, 
But normal api requests `fetch('/api/test')` may fail due to a change in language that causes the URL to change `fetch('/api/[locale]/test')`.
Of course, I can fix it by doing the following, but I doubt if this is a good way.
```
  const host = headers().get("host");
  const protocal = process?.env.NODE_ENV === "development" ? "http" : "https";
  const res = await fetch(`${protocal}://${host}/api`);
```

Has anyone else had this problem? How did you solve it?
I'd like to find a solution where the folder structure looks like the following and both Next-Auth and Fetch work properly.
`src/app/[locale]`
`src/app/api`

**Comment by sichenzhang126** [comment]
Encountered the same problem here. 
What I got is "/[locale]/api/xxx" instead of "/api/[locale]/xxx" as you mentioned. 
Wonder if you got this solved?

**Comment by thallysondias** [comment]
add api in middleware:
**!api**


```
import createMiddleware from "next-intl/middleware";
import { routing } from "./i18n/routing";

export default createMiddleware(routing);

export const config = {
  // Exclui rotas da API e de arquivos internos do Next.js como _next e _vercel
  matcher: [
    "/((?!api|_next|_vercel|.*\\..*).*)",
    "/(pt|en|es)/:path*"
  ],
}
```
